src/plugins/new-ui/public/webcomponents/power.html

```diff
@@ -216,28 +169,58 @@
           
           element.style.transform=("translate3d(" + config.X + "px," + val + "px,0px )");
         },
-        setVoltage: function (voltage) {
+
+        hslColorFromPercent: function( value )
+        {
+            // Maps a decimal percentage 0.0:1.0 to red:green hues
+            var hue = ( ( value ) * 120 ).toString( 10 );
+
+            // Saturation and Lightness are locked
+            return [ "hsl(",hue,",90%,40%)" ].join("");
+        },
+
+        setVoltage: function (voltage) 
+        {
           voltage = parseFloat(voltage);
+
           this.set("voltage",voltage);
-          var change = (this.maxVoltage - this.minVoltage) / (voltage - this.minVoltage);
-          var percentage = 100 / change;
-          var indicatorWidth = 200 / change;
-          indicatorWidth = indicatorWidth.toFixed(2);
-          var fillColor = undefined;
-          this.colors.forEach(function (color) {
-            if (fillColor === undefined && percentage <= color.value) {
-              fillColor = color.color;
+
+          var percent;
+
+          // Apply floor and ceiling to voltages which fall outside the expected range for the selected battery chemistry
+          if( voltage <= this.minVoltage )
+          {
+            percent = 0.0;
+          }
+          else if( voltage < this.maxVoltage )
+          {
+            percent = (voltage - this.minVoltage) / (this.maxVoltage - this.minVoltage);
+
+            // Handle cases where min and max are equal, causing divide-by-0
+            if( !isFinite( percent ) )
+            {
+              percent = 0.0;
             }
-          });
-          if (fillColor == undefined) {
-            fillColor = this.colors[this.colors.length - 1].color;
           }
-          ;
-          if (!isNaN(indicatorWidth) && isFinite( indicatorWidth ) && indicatorWidth >= 0) {
-            $(this.$.indicator).css({ fill: fillColor });
-            $(this.$.indicator).attr('width', indicatorWidth);
+          else
+          {
+            percent = 1.0;
           }
+
+          // Calculate fill width
+          var fillWidth  = ( percent * 200.0 ).toFixed(2);
+
```

Comment:
Added a minimum fill width so the user has some visual color feedback about zero or below-zero voltages